### ISSUE TITLE
Boosted heatmap series not rendered in 11.3.0

### ISSUE THREAD

**Issue by jakubjanuchta** [highcharts/highcharts#20456]
#### Expected behaviour
Boosted heatmap series should be rendered

#### Actual behaviour
Boosted heatmap series is not rendering

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/umh54k23/

#### Product version
Highcharts v11.3.0

#### Last working version
Highcharts v11.2.0

#### Affected browser(s)
All

_Internal note:_
Bisected to: https://github.com/highcharts/highcharts/commit/a67ba29717127638d1e1c742b16bcb6f50e25986

**Comment by TorsteinHonsi** [comment]
This one appears to get fixed by https://github.com/highcharts/highcharts/pull/20449.

See the [demo with the fix applied](https://jsfiddle.net/highcharts/8xno3wp0/).

**Comment by kamil-musialowski** [comment]
It seems that it's still failing for datasets above 1000 points, could you guys confirm? 
Demo with the fix from #20449 : https://jsfiddle.net/BlackLabel/7yoq4dgs/

**Comment by hubertkozik** [comment]
For more than 1000 points, the chart is not showing up.

**Demo with code directly from branch:** https://jsfiddle.net/BlackLabel/ao24mb6p/

**Comment by reydoblas** [comment]
Any update on this? The suggested workaround is still failing for more than 1000 points.

**Comment by hubertkozik** [comment]
Hi @reydoblas! The ticket is prioritized in our backlog and as you can see it is classified as a blocker. When the work starts you'll get the notifications here.

**Comment by reydoblas** [comment]
@hubertkozik Any update on this? There's a new release version 11.4.0 but the fix for this is not yet there. I can see 

#Fixed [#20433](https://github.com/highcharts/highcharts/issues/20433), boosted scatter chart did not work without axis min/max-options.

as part of the changelog but it still don't fix the heatmap series not rendering in boost mode.

**Comment by hubertkozik** [comment]
Hi @reydoblas!
This issue is not fixed yet. Indeed, there was a similar issue with the boosted scatter series, but this issue is about boosted heatmap. @MarkusBarstad is working on this issue.

**Comment by MarkusBarstad** [comment]
@reydoblas [Here is a workaround you can try](https://jsfiddle.net/mkb93/tmckyLgv/)

**Comment by ahansen1** [comment]
In case it is helpful to anyone, I tried the above workaround from @MarkusBarstad, but was still getting an exception on 11.3.0+ (it worked in 11.2.0) from hasData and the heatmap would not render.   I was able to work around the exception by replacing the hasData which uses xData instead of processedXData.  Not sure this is functionally correct, but it resolves the issue.

```javascript
(function(H) {
  H.wrap(H.Series.types.heatmap.prototype, 'hasData', function(proceed) {
    return !!this?.xData?.length
  });
})(Highcharts);
``` 


Here is a fiddle that demonstrates the issue: https://jsfiddle.net/x0jc3qrn/1/
